fbw-a32nx/src/systems/atsu/src/AOC.ts

```diff
@@ -33,6 +33,10 @@ export class Aoc {
     public removeMessage(uid: number): boolean {
         const index = this.messageQueue.findIndex((element) => element.UniqueMessageID === uid);
         if (index !== -1) {
+            if (this.messageQueue[index].Confirmed === false) {
+                const cMsgCnt = SimVar.GetSimVarValue('L:A32NX_COMPANY_MSG_COUNT', 'Number');
+                SimVar.SetSimVarValue('L:A32NX_COMPANY_MSG_COUNT', 'Number', cMsgCnt <= 1 ? 0 : cMsgCnt - 1);
+            }
```

Comment:
I suggest a less error-prone patch, add a new method called updateMessageCount or something and call it in each of the functions that mutates the queue (better would be to wrap the queue in a nice abstraction but that's an even bigger patch). Probably also call it in an init function that's called (indirectly) from onConnectedCallback in the instrument so it clears the count when the instrument is reloaded etc.
```ts
private async updateMessageCount(): Promise<unknown> {
    const count = this.messageQueue.reduce((c, m) => m.Confirmed ? c : c + 1, 0);
    return SimVar.SetSimVarValue('L:A32NX_COMPANY_MSG_COUNT', 'number', count);
}
```

Reply:
We can go for the easiest fix.
Will be replaced with the a new logic in next ATSU update in my PR

Reply:
Fair enough, good suggestion. The code duplication did cross my mind, but I wasn't sure if it was worth a new function.